### PR TITLE
Authentication functionality with note page setup

### PR DESCRIPTION
**Pull request by dev-enforced**
In this Pull Request one can find the authentication functionality implemented on the LOGIN and signup screens 
On successful login and signup one is redirected to the home page of the notes app which for now has 2 major components one for taking the input to form a new note and other for displaying the notes available. The views available on the notes Home aren't functional yet and are just for the viewing part and the styling will be improved in the upcoming Prs
Here is the live preview of the website: https://notes-app-recreated-git-authentication-func-bf1088-dev-enforced.vercel.app/

### CODE REVIEW

File: src/components/ColorPalette/ColorPalette.css [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+.color-list{
+    width: 14rem;
+    top:2.5rem;
+    z-index: 2;
+    position: absolute;
+    border-radius: var(--border-rd-md);
+    gap:1rem;
+    padding: 0.7rem 1rem;
+    background-color: #253053;
+    box-shadow: 1px 2px 15px #d1d1d1;
```
Comment (t007rushi) [inline src/components/ColorPalette/ColorPalette.css:16]: Use variables

File: src/components/ColorPalette/ColorPalette.jsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import React, { useState } from 'react';
+import { colorList } from 'constants';
+import "./ColorPalette.css";
+const ColorPalette = () => {
+    const [colorListStyle, setColorListStyle] = useState("color-list-none")
+    const toggleColorListStyle = () => {
+        setColorListStyle(classDetails => classDetails === "color-list flex-row" ? setColorListStyle("color-list-none") : ("color-list flex-row"))
```
Comment (t007rushi) [inline src/components/ColorPalette/ColorPalette.jsx]: What is this? Setting value inside another set?🤔

File: src/components/ColorPalette/ColorPalette.jsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+            <button className='color-selector-btn' onClick={toggleColorListStyle}>
+                <span className="material-icons-outlined app-icon">
+                    palette
+                </span>
+            </button>
+            <div className={colorListStyle}>
+                {
+                    colorList.map((everyColor, index) => {
+                        return (
+                            <span className='color-element' key={index} style={{ backgroundColor: everyColor }} value={everyColor}></span>
```
Comment (t007rushi) [inline src/components/ColorPalette/ColorPalette.jsx]: Avoid inline styling 
Instead create diff classes in css
And then match everyColor with that class
```ClassName={curr color?"style class":"default"}```

File: src/components/HomePageNavigation/HomePageNavigation.jsx [hunk 4]
```diff
@@ -11,9 +17,16 @@ const HomePageNavigation = () => {
...
-          <NavLink className={`home-link link-none `} to="/login">
-            LOGIN
-          </NavLink>
+          {isLoggedIn ?
+            <button className={`hero-btn hero-btn-primary py-2 px-3`} onClick={() => {
+              logoutService(navigate, setAuthState);
+            }} >
+              LOGOUT
+            </button> :
+            <NavLink className={`home-link link-none`} to="/login">
```
Comment (t007rushi) [inline src/components/HomePageNavigation/HomePageNavigation.jsx:26]: Remove deafult active class styling as it shows underline decoration

File: src/components/HomePageNavigation/HomePageNavigation.jsx [hunk 5]
```diff
@@ -11,9 +17,16 @@ const HomePageNavigation = () => {
       </Link>
       <ul className="home-links flex-row flex-align-center">
         <li>
-          <NavLink className={`home-link link-none `} to="/login">
-            LOGIN
-          </NavLink>
+          {isLoggedIn ?
+            <button className={`hero-btn hero-btn-primary py-2 px-3`} onClick={() => {
+              logoutService(navigate, setAuthState);
```
Comment (t007rushi) [inline src/components/HomePageNavigation/HomePageNavigation.jsx:22]: Don't pass setter function into any function or navigate never

File: src/components/NoteItem/NoteItem.css [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+}
+
+.card-actions .btn{
+    border-radius: 50%;
+    font-size: 1.2rem;
+}
+
+.card-actions .btn:hover{
+    color:white;
+    background-color:#d311e1;
```
Comment (t007rushi) [inline src/components/NoteItem/NoteItem.css:12]: Look for better bg color as text not readable

File: src/components/NoteItem/NoteItem.css [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+    background: transparent;
+}
+
+.card-actions .btn{
+    border-radius: 50%;
+    font-size: 1.2rem;
+}
+
+.card-actions .btn:hover{
+    color:white;
```
Comment (t007rushi) [inline src/components/NoteItem/NoteItem.css:11]: - And use color variable here

File: src/services/Authentication/getLoginResponse.js [hunk 8]
```diff
@@ -0,0 +1,12 @@
+import axios from "axios";
+const getLoginResponse = async ({ email, password }) => {
+    try {
+        const loginResponse = await axios.post("/api/auth/login", {
```
Comment (t007rushi) [inline src/services/Authentication/getLoginResponse.js:4]: Can destructure data here only

**Review by t007rushi — COMMENTED** [review]
Look for IMP changes

File: src/services/Authentication/getLoginResponse.js (hunk 8 above)
Comment (dev-enforced) [reply src/services/Authentication/getLoginResponse.js:4]: Have destructured it while writing the submission handler function for Login and Signup operations. Here I intend to collect the whole response whenever Signup and Login happens.

File: src/components/ColorPalette/ColorPalette.jsx (hunk 2 above)
Comment (dev-enforced) [reply src/components/ColorPalette/ColorPalette.jsx]: Yes missed it !! Thanks for pointing it out!!

**Review by dikshadhote — APPROVED** [review]
Rushikesh has given all valid points that you can work on. Just One suggestion from my side.Make join now as CTA. It seems like just text.just change text color on hover or may be give it a underline style.
Also you can use cases like this ""New to Unote? JOIN NOW 
image

File: src/components/ColorPalette/ColorPalette.jsx (hunk 3 above)
Comment (dev-enforced) [reply src/components/ColorPalette/ColorPalette.jsx]: Made the changes related to it using the index property

File: src/components/NoteItem/NoteItem.css (hunk 6 above)
Comment (dev-enforced) [reply src/components/NoteItem/NoteItem.css:12]: Yes will work on it in the next PR while working on add To Notes !!

File: src/components/HomePageNavigation/HomePageNavigation.jsx (hunk 4 above)
Comment (dev-enforced) [reply src/components/HomePageNavigation/HomePageNavigation.jsx:26]: Didn't understand what you are trying to convey!!Can you show it through a snapshot?

**Pull request merged**